Add orca DAAC module configuration hooks into CIRRUS-core

Jkovarik opened the pull request:
This PR adds a new Core output from v17 that's needed to integrate Orca with CIRRUS.

mattp0 reviewed (commented):
Looks good to me

lindsleycj commented:
@Jkovarik one very small nit.  In the new version, I have generally always used the Cumulus version for the first three levels and use the 4th level for a change to CIRRUS only (which this change is).  We're still at v17.0.0 of Cumulus.  So, if you don't mind, please change the version in the CHANGELOG to `v17.0.0.1`.  Sound ok?  I'm glad to merge and make the release after that.

Jkovarik commented:
@lindsleycj  will do, thanks for the comment!   I discovered an implementation issue, will ping you once I've updated the PR.   More to come.

Jkovarik commented:
@lindsleycj  I updated the PR to address your comment.      bce8af8 

Additionally after working a bit more on the downstream effort I updated the PR to add additional functionality - I'd be happy to walk through the changes and why, but will add context here as well.

Jkovarik commented:
The new updates are intended to address the current scenario introduced by Cumulus v17, where Core now provides the ORCA workflow lambdas, but in doing so requires configuration inputs from the Orca module.  

That means DAACs/users that are deploying with Orca need to deploy in the following order: 
```
	tf
	daac
	rds
	<<orca>>
	data-persistence
	cumulus
	workflows
```
As orca needs *input* from daac (in the form of the buckets), as well as rds (or presumably another module if the orca database is not co-located in the same database cluster), and provide configuration *to* cumulus. 

For the moment, we're targeting add add another PR to the cirrus-daac repo to provide the `orca` optional model.     Would love feedback/suggestions/discussions of the approach.   Thanks!

Jkovarik commented on Makefile:
```diff
@@ -192,7 +192,6 @@ cumulus: cumulus-init
 	fi
 	export TF_CMD="terraform apply \
 				-var-file=${DAAC_DIR}/$@/terraform.tfvars \
-				$$VARIABLES_OPT \
```
This shouldn't be removed.   Will fix.

lindsleycj commented on cumulus/common.tf:
@mattp0 @reweeden @mckadesorensen @mikedorfman Hi all.  Am I the only `CIRRUS` DAAC that doesn't use ORCA yet? Per this PR and my question above, if I'm the only one not using ORCA, I could go ahead and do a deployment of it such that the remote state does exist.

reweeden replied:
ASF hasn't used ORCA yet either.

Jkovarik replied:
@lindsleycj    I realized the new orca.tf wasn't included in the push, I've remedied that - the remote state is conditional on having `use_orca` set to true.      

The `orca_remote_state_config` block is text only, and all other actionable values (e.g. orca_lambda_copy_to_archive) and/or the remote state are conditional. 

If you *do* have use orca set to true, you must deploy Orca before Cumulus, but after daac/rds - this will break using the default Makefile, but that's implicit in having an orca module that isn't part of the cirrus-core module that has hooks like Orca does. 

I'd be happy to talk over why in more depth if that makes sense.

Jkovarik commented on cumulus/orca.tf:
```diff
@@ -0,0 +1,7 @@
+
+data "terraform_remote_state" "orca" {
+  count = var.use_orca == true ? 1 : 0
```
@lindsleycj  this is the conditional expression I was referring to above

Jkovarik replied:
This is outdated - I merged main and added this to the data.tf manifest

Jkovarik commented on CHANGELOG.md:
@lindsleycj  Can do, will update.  Thanks for taking a look!

Jkovarik replied:
@lindsleycj  apologies for missing that merge issue - I've updated it 791cb28

lindsleycj commented:
@Jkovarik I merged in PR #172 to master so I could go ahead and create the v17.0.0.1 release.  I then fixed the conflict in this PR.

lindsleycj reviewed (approved):
I wanted to make sure this worked for people who don't have ORCA installed.  

I checked out this branch and did a `make cumulus` for a sandbox deployment that was already on v17.0.0 of cumulus, it worked fine.  So, I approve!  

I can't check it for those that really do have ORCA, so I'll leave that for someone else

mikedorfman commented:
@lindsleycj - this looks reasonable to me as well.  NSIDC hasn't used ORCA yet, but we will very soon in this PI.  Thanks for testing it in your sandbox!

Jkovarik commented:
@lindsleycj  appreciate the review/comments!

mattp0 reviewed (approved):
Looks good!
